core/Makefile

```diff
@@ -75,6 +78,28 @@ else ifeq ($(words $(MAKECMDGOALS)),1)
 	endif
 endif
 
+# We always want these libraries to be dynamically linked even when the
+# user requests a static build.  These needs to be listed first, so even
+# if they're listed again in the static section, they will be dynamically
+# linked.
+ALWAYS_DYN_LIBS := -lpthread -ldl
```

Comment:
Is it necessary to add `-lpthread` here? The `-pthread` option already implies this...

Reply:
I believe it's necessary.

grpc includes `-lpthread` in its statically required libraries.  If we don't specify `-lpthread` before (in the dynamic section), -lpthread will be linked statically, and that's not well supported apparently